ui/com/composer/index.jsx

```diff
@@ -203,6 +203,314 @@ export default class Composer extends React.Component {
     return !!this.state.text.trim()
   }
 
+  // onSend()
+
+  // Stateless components
+  renderComposerTextarea(vertical) {
+    return (vertical ? ComposerTextareaVerticalFilled : ComposerTextareaFixed)
+  }
+
+  renderPreview(props) {
+    return (props) => (
+      <div>
+        <div className="card" style={{padding: '20px', margin: '40px 10px 30px 0'}}>
+          <MarkdownBlock md={this.state.text} />
+        </div>
+      </div>)
+  }
+
+  renderAudienceBtn(props) {
+    return (props) => {
+      const opts = [
+        { label: <span><i className="fa fa-inbox"/> Private</span>, value: false },
+        { label: <span><i className="fa fa-bullhorn"/> Public</span>, value: true }
+      ]
+      if (!props.canChange) {
+        return (<a className="btn disabled">{opts[+props.isPublic].label}</a>)
+      } else {
+        return (<DropdownBtn className="btn" items={opts} onSelect={props.onSelect}>
+        {opts[+props.isPublic].label} <i className="fa fa-caret-down" />
+        </DropdownBtn>)
+      }
+    }
+  }
+
+  renderAttachBtn(props) {
+    return (props) => {
+      if (!props.isPublic) {
+        if (props.isReply) {
+          return <span/>
+        } else {
+          return <a className="btn disabled">
+          <i className="fa fa-paperclip" /> Attachments not available in PMs</a>
```

Comment:
Here's another wonky indentation place. I'd prefer this looked like:

```js
return <a className="btn disabled">
  <i className="fa fa-paperclip" /> Attachments not available in PMs
</a>
```

Or was all on one line